BUG: 小红书解析出错

## 版本
1.9.3

## 解析地址

实际上消息是个卡片，这是只复制出来了分享链接

`https://www.xiaohongshu.com/discovery/item/685fd0e00000000024008b56?app_platform=android&ignoreEngage=true&app_version=8.87.6&share_from_user_hidden=true&xsec_source=app_share&type=video&xsec_token=CBc7kDk5WA32hs6hpCZ4jOhP1n0l8OeJ0kOeeUOoEHPl8%3D&author_share=1&xhsshare=QQ&shareRedId=N0w7NTk7ND82NzUyOTgwNjY0OTc4Sz9N&apptime=1751343431&share_id=c644022d3b18407d95807a10b14f0658&share_channel=qq&qq_aio_chat_type=2`

## 日志
```
07-01 12:17:16 [ERROR] nonebot | Running Matcher(type='message', module=nonebot_plugins.nonebot_plugin_resolver2.matchers.xiaohongshu, lineno=14) failed.
Traceback (most recent call last):

  File "E:\Bot\bot.py", line 34, in <module>
    nonebot.run()
    │       └ <function run at 0x00000210461794E0>
    └ <module 'nonebot' from 'C:\\Users\\Administrator\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\zhenxun-bot-oFP0POfx-py3.11\\...

  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\__init__.py", line 337, in run
    get_driver().run(*args, **kwargs)
    │                 │       └ {}
    │                 └ ()
    └ <function get_driver at 0x000002104539A3E0>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\drivers\fastapi.py", line 187, in run
    uvicorn.run(
    │       └ <function run at 0x0000021047A534C0>
    └ <module 'uvicorn' from 'C:\\Users\\Administrator\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\zhenxun-bot-oFP0POfx-py3.11\\...
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\uvicorn\main.py", line 579, in run
    server.run()
    │      └ <function Server.run at 0x0000021047A21B20>
    └ <uvicorn.server.Server object at 0x0000021059FC1850>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ None
           │       │   │    └ <function Server.serve at 0x0000021047A21BC0>
           │       │   └ <uvicorn.server.Server object at 0x0000021059FC1850>
           │       └ <function run at 0x0000021044D7D8A0>
           └ <module 'asyncio' from 'C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x000002105D533C40>
           │      └ <function Runner.run at 0x0000021045258CC0>
           └ <asyncio.runners.Runner object at 0x000002105D58B650>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\z...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000021045256840>
           │    └ <ProactorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x000002105D58B650>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
    │    └ <function ProactorEventLoop.run_forever at 0x00000210452BFC40>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x00000210452585E0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1922, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000021044D5CF40>
    └ <Handle Task.task_wakeup(<Task finishe...> result=None>)>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Task finishe...> result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Task finishe...> result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Task finishe...> result=None>)>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\utils.py", line 254, in run_coro_with_shield
    return await coro
                 └ <coroutine object check_and_run_matcher at 0x0000021002460040>
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\message.py", line 506, in check_and_run_matcher
    await _run_matcher(
          └ <function _run_matcher at 0x00000210462F8F40>
> File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\message.py", line 458, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
          │       │   │    │      │      │      └ {<function _event_to_me at 0x000002104610BEC0>: <nonebot.internal.params.DependencyCache object at 0x000002100243F0D0>, <func...
          │       │   │    │      │      └ <contextlib.AsyncExitStack object at 0x000002105F16D5D0>
          │       │   │    │      └ {'_r_extract': '[分享]ASMR🎧pick哪只猫咪捏捏\n#asmr  #asmr助眠  #asmr治愈  #asmr解压助眠  #猫咪捏…\nhttps://www.xiaohongshu.com/discovery/item/68...
          │       │   │    └ GroupMessageEvent(time=1751343435, self_id=0000000000, post_type='message', sub_type='normal', user_id=3067641715, message_ty...
          │       │   └ Bot(type='OneBot V11', self_id='0000000000')
          │       └ <function Matcher.run at 0x0000021046108720>
          └ Matcher(type='message', module=nonebot_plugins.nonebot_plugin_resolver2.matchers.xiaohongshu, lineno=14)
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 926, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
          │    │          │    │      │      │      └ {<function _event_to_me at 0x000002104610BEC0>: <nonebot.internal.params.DependencyCache object at 0x000002100243F0D0>, <func...
          │    │          │    │      │      └ <contextlib.AsyncExitStack object at 0x000002105F16D5D0>
          │    │          │    │      └ {'_r_extract': '[分享]ASMR🎧pick哪只猫咪捏捏\n#asmr  #asmr助眠  #asmr治愈  #asmr解压助眠  #猫咪捏…\nhttps://www.xiaohongshu.com/discovery/item/68...
          │    │          │    └ GroupMessageEvent(time=1751343435, self_id=0000000000, post_type='message', sub_type='normal', user_id=3067641715, message_ty...
          │    │          └ Bot(type='OneBot V11', self_id='0000000000')
          │    └ <function Matcher.simple_run at 0x0000021046108680>
          └ Matcher(type='message', module=nonebot_plugins.nonebot_plugin_resolver2.matchers.xiaohongshu, lineno=14)
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 863, in simple_run
    await handler(
          └ Dependent(call=_)
  File "C:\Users\Administrator\AppData\Local\pypoetry\Cache\virtualenvs\zhenxun-bot-oFP0POfx-py3.11\Lib\site-packages\nonebot\dependencies\__init__.py", line 113, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
                 │    │             │         │    │    │       └ {'text': '[分享]ASMR🎧pick哪只猫咪捏捏\n#asmr  #asmr助眠  #asmr治愈  #asmr解压助眠  #猫咪捏…\nhttps://www.xiaohongshu.com/discovery/item/685fd0e0...
                 │    │             │         │    │    └ <function _ at 0x000002105A2700E0>
                 │    │             │         │    └ Dependent(call=_)
                 │    │             │         └ ~R
                 │    │             └ <class 'collections.abc.Awaitable'>
                 │    └ typing.Callable
                 └ <function cast at 0x00000210446FA840>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_resolver2\exception.py", line 31, in wrapper
    return await func(*args, **kwargs)
                 │     │       └ {'text': '[分享]ASMR🎧pick哪只猫咪捏捏\n#asmr  #asmr助眠  #asmr治愈  #asmr解压助眠  #猫咪捏…\nhttps://www.xiaohongshu.com/discovery/item/685fd0e0...
                 │     └ ()
                 └ <function _ at 0x000002105A270040>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_resolver2\matchers\xiaohongshu.py", line 28, in _
    parse_result = await xhs_parser.parse_url(matched.group(0))
                         │          │         │       └ <method 'group' of 're.Match' objects>
                         │          │         └ <re.Match object; span=(62, 457), match='https://www.xiaohongshu.com/discovery/item/685fd0>
                         │          └ <function XiaoHongShuParser.parse_url at 0x0000021059FD5940>
                         └ <nonebot_plugins.nonebot_plugin_resolver2.parsers.xiaohongshu.XiaoHongShuParser object at 0x000002105A1E8450>

  File "E:\Bot\nonebot_plugins\nonebot_plugin_resolver2\parsers\xiaohongshu.py", line 84, in parse_url
    video_url = note_data["video"]["media"]["stream"]["h264"][0]["masterUrl"]
                └ {'xsecToken'xxx', 'desc': '#asmr[话题]# #asmr助眠[话题]# #asmr治愈[话题]# #asmr解压助眠[话题]# ...

IndexError: list index out of range
```

--- a/nonebot_plugin_resolver2/parsers/xiaohongshu.py
+++ b/nonebot_plugin_resolver2/parsers/xiaohongshu.py
@@ -81,7 +81,13 @@ class XiaoHongShuParser:
             image_list = note_data["imageList"]
             img_urls = [item["urlDefault"] for item in image_list]
         elif resource_type == "video":
-            video_url = note_data["video"]["media"]["stream"]["h264"][0]["masterUrl"]
+            stream = note_data["video"]["media"]["stream"]
+            for code in ("h264", "h265", "av1"):
+                if item := stream.get(code):
+                    video_url = item[0]["masterUrl"]
+                    break
+            if not video_url:
+                raise ParseException("小红书视频解析失败")
         else:
             raise ParseException(f"不支持的小红书链接类型: {resource_type}")
         return ParseResult(

--- a/tests/test_xhs.py
+++ b/tests/test_xhs.py
@@ -14,6 +14,7 @@ async def test_xiaohongshu():
     urls = [
         "https://www.xiaohongshu.com/discovery/item/67cdaecd000000000b0153f8?source=webshare&xhsshare=pc_web&xsec_token=ABTvdTfbnDYQGDDB-aS-b3qgxOzsq22vIUcGzW6N5j8eQ=&xsec_source=pc_share",
         "https://www.xiaohongshu.com/explore/67ebf78f000000001c0050a1?app_platform=ios&app_version=8.77&share_from_user_hidden=true&xsec_source=app_share&type=normal&xsec_token=CBUGDKBemo2y6D0IIli9maqDaaazIQjzPrk2BVRi0FqLk=&author_share=1&xhsshare=QQ&shareRedId=N0pIOUc1PDk2NzUyOTgwNjY0OTdFNktO&apptime=1744081452&share_id=00207b217b7b472588141b083af74c7a",
+        "https://www.xiaohongshu.com/discovery/item/685fd0e00000000024008b56?app_platform=android&ignoreEngage=true&app_version=8.87.6&share_from_user_hidden=true&xsec_source=app_share&type=video&xsec_token=CBc7kDk5WA32hs6hpCZ4jOhP1n0l8OeJ0kOeeUOoEHPl8%3D&author_share=1&xhsshare=QQ&shareRedId=N0w7NTk7ND82NzUyOTgwNjY0OTc4Sz9N&apptime=1751343431&share_id=c644022d3b18407d95807a10b14f0658&share_channel=qq&qq_aio_chat_type=2",
     ]
 
     async def test_parse_url(url: str) -> None:

